packtools/sps/formats/pubmed.py

```diff
@@ -703,13 +679,21 @@ def xml_pubmed_other_abstract(xml_pubmed, xml_tree):
       <AbstractText Label="CONCLUSION">Une meilleure estimation du fardeau économique global des facteurs de risque multiples au sein d’une population peut faciliter l’établissement des priorités et améliorer le soutien aux initiatives de prevention primaire. </AbstractText>
     </OtherAbstract>
     """
-    abstract_el = ET.Element("OtherAbstract")
-    for lang, abstract in get_abstracts(xml_tree).items():
-        if lang != "en":
-            abstract_el.set("Language", lang)
-            if not abstract["structured"]:
-                abstract_el.text = abstract.get("text")
-            else:
-                for item in abstract.get("text"):
-                    abstract_el.append(add_abstract_text(item.get("title"), item.get("p")))
-            xml_pubmed.append(abstract_el)
+    try:
+        main_lang = article_abstract.Abstract(xml_tree).get_main_abstract().get('lang')
+        abstracts = article_abstract.Abstract(xml_tree).get_abstracts()
+        for abstract, lang in [(item.get('abstract'), item.get('lang')) for item in abstracts]:
```

Comment:
Esta usando for duas vezes sem necessidade. A linha 685, é o mesmo que:

```python
for item in abstracts:
    abstract = item.get('abstract')
    lang = item.get('lang')
```
Não economiza linha, mas economiza processamento, que não é 2 vezes o for, mas N vezes N.

Reply:
OK, @robertatakenaka